src/overlays/actors/ovl_En_Az/z_en_az.c

```diff
@@ -39,108 +80,1762 @@ const ActorInit En_Az_InitVars = {
+    actorPathing->moveFunc = SubS_ActorPathing_MoveWithGravity;
+    if (actorPathing->distSqToCurPointXZ <= this->actor.speedXZ) {
+        ret = true;
+    }
+    return ret;
+}
+
+s32 func_80A9565C(PlayState* play, ActorPathing* actorPathing) {
+    EnAz* this = (EnAz*)actorPathing->actor;
+    s32 ret = false;
+    f32 temp_f0;
 
-extern ColliderCylinderInit D_80A99110;
+    this->actor.gravity = -1.0f;
+    actorPathing->moveFunc = func_80A94A90;
+    this->unk_374 |= 0x2000;
+    temp_f0 = func_80A954AC(this);
+    if ((actorPathing->distSqToCurPointXZ < SQ(this->actor.speedXZ)) || (temp_f0 <= 0.0f)) {
+        ret = true;
+    } else {
+        this->unk_39E = this->actor.world.rot.x =
+            Math_Atan2S(-this->actor.velocity.y, Math_CosS(-this->actor.world.rot.x) * this->actor.speedXZ);
+    }
+    return ret;
+}
 
-extern UNK_TYPE D_0600C94C;
-extern UNK_TYPE D_0601ABF0;
+s32 func_80A95730(PlayState* play, ActorPathing* actorPathing) {
+    EnAz* this = (EnAz*)actorPathing->actor;
+    s32 ret = false;
+    f32 temp_f0;
+    f32 sp40;
+    s32 sp3C;
+    s32 sp38;
+    s32 sp34;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Az/func_80A94A30.s")
+    this->actor.gravity = 0.0f;
+    temp_f0 = func_80A954AC(this);
+    if ((actorPathing->distSqToCurPointXZ < SQ(this->actor.speedXZ)) || (temp_f0 <= 0.0f)) {
+        ret = true;
+    } else {
+        sp40 = SQ(this->actor.speedXZ) / actorPathing->distSqToCurPoint;
+        sp34 = ABS(actorPathing->rotToCurPoint.x - this->actor.world.rot.x);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Az/func_80A94A64.s")
+        sp3C = (s32)(sp34 * sp40) + 0xAAA;
+        sp34 = ABS(actorPathing->rotToCurPoint.y - this->actor.world.rot.y);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Az/func_80A94A90.s")
+        Math_SmoothStepToS(&this->actor.world.rot.x, actorPathing->rotToCurPoint.x, 1, sp3C, 0);
+        sp38 = (s32)(sp34 * sp40) + 0xAAA;
+        Math_SmoothStepToS(&this->actor.world.rot.y, actorPathing->rotToCurPoint.y, 1, sp38, 0);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Az/func_80A94AB8.s")
+        Math_SmoothStepToS(&this->unk_39E, this->actor.world.rot.x, 2, sp3C, 0);
+        Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.world.rot.y, 2, sp38, 0);
+    }
+    actorPathing->moveFunc = SubS_ActorPathing_MoveWithoutGravityReverse;
+    return ret;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Az/func_80A94B20.s")
+s32 func_80A958B0(PlayState* play, ActorPathing* actorPathing) {
+    EnAz* this = (EnAz*)actorPathing->actor;
+    s32 ret = false;
+    f32 temp;
+    f32 phi_f0;
+    f32 temp1;
+    f32 temp2;
+    f32 sp3C;
+    s32 sp2C;
+    s32 sp28;
+    s32 sp30;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Az/func_80A94B98.s")
+    this->actor.gravity = 0.0f;
+    temp2 = D_80A99E8C;
+    temp = D_80A99E90[this->unk_2F8];
```

Comment:
Is `D_80A99E90` never set? Is this undefined behaviour?

Reply:
It's set in `func_80A97F9C`, but I don't understand the actor well enough to know if that always runs before this one. It wouldn't be the first time that zero-initialization of bss is assumed, and I'm not sure if that's undefined.